runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/content/FileContent.kt

```diff
@@ -21,5 +22,43 @@ public class FileContent(
     override val contentLength: Long
         get() = endInclusive - start + 1
 
-    override fun newReader(): SdkByteReadChannel = file.readChannel(start, endInclusive)
+    override fun newReader(): SdkByteReadChannel {
+        val ch = FileChannel()
+        val job = CoroutineScope(Dispatchers.IO + CoroutineName("file-reader")).launch {
+            file.source(start, endInclusive).use {
+                ch.writeAll(it)
+            }
+        }
```

Comment:
When newReader is called it starts reading

Reply:
But `writeAll` stops after pushing some fixed amount and doesn't slurp up the whole file at once?

Reply:
The amount of data that can be written to a channel is constrained by the maximum channel buffer size. So yes it would suspend after it fills the buffer until a reader consumes some data making additional write capacity available.

Reply:
FWIW I don't love this. This is how ktor file channels worked as well and it warrants investigating consuming an `SdkSource` directly but updating the channel API is a smaller incremental change than adding an entire new path for HTTP engines to consider.